File: api-gateway/src/main/java/com/netflix/zuul/config/FiltersConfig.java
package com.netflix.zuul.config;

import com.netflix.zuul.filters.pre.DebugRequestFilter;
import com.netflix.zuul.filters.pre.PreDecorationFilter;
import com.netflix.zuul.filters.route.SimpleHostRoutingFilter;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class FiltersConfig {

//    @Bean
//    public SimpleHostRoutingFilter simpleHostRoutingFilter() {
//        return new SimpleHostRoutingFilter();
//    }

    @Bean
    public PreDecorationFilter preDecorationFilter() {
        return new PreDecorationFilter();
    }

    @Bean
    public DebugRequestFilter debugRequestFilter() {
        return new DebugRequestFilter();
    }

}

File: api-gateway/src/main/kotlin/com/bmrobin/zuul/filters/config/FiltersConfig.kt
package com.bmrobin.zuul.filters.config

import com.bmrobin.zuul.filters.post.AddResponseHeaderFilter
import com.bmrobin.zuul.filters.pre.ServiceRequestFilter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class FiltersConfig {

    @Bean
    fun serviceRequestFilter(): ServiceRequestFilter {
        return ServiceRequestFilter()
    }

    @Bean
    fun addResponseHeaderFilter(): AddResponseHeaderFilter {
        return AddResponseHeaderFilter()
    }

}
